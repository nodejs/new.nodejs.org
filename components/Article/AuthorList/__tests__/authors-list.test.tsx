import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import AuthorsList from '..';

describe('AuthorsList component', () => {
  it('renders correctly', () => {
    const authors = ['test-author', 'another-test-author'];
    const { container } = render(
      <IntlProvider locale="en" onError={() => {}}>
        <AuthorsList authors={authors} />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
