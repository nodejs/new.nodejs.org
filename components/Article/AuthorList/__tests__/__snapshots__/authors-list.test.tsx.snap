// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsList component renders correctly 1`] = `
<div>
  <div
    class="authorList"
  >
    components.article.authorList.title
    <ul>
      <li>
        <a
          aria-label="components.article.author.githubLinkLabel"
          class="link"
          href="https://github.com/test-author"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="test-author"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="http://localhost/_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D60&w=128&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D60&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D60&w=128&q=75 2x"
            style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            width="60"
          />
        </a>
      </li>
      <li>
        <a
          aria-label="components.article.author.githubLinkLabel"
          class="link"
          href="https://github.com/another-test-author"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="another-test-author"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="http://localhost/_next/image?url=https%3A%2F%2Fgithub.com%2Fanother-test-author.png%3Fsize%3D60&w=128&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Fanother-test-author.png%3Fsize%3D60&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Fanother-test-author.png%3Fsize%3D60&w=128&q=75 2x"
            style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            width="60"
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
